sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -373,3 +406,87 @@ def get_region_zones_for_accelerators(
     """Returns a list of regions for a given accelerators."""
     df = _get_accelerator(_df, accelerator, count, region=None)
     return common.get_region_zones(df, use_spot)
+
+
+def check_host_accelerator_compatibility(instance_type: str,
+                                         accelerators: Optional[Dict[str, int]],
+                                         zone: Optional[str] = None) -> None:
+    """Check if the instance type is compatible with the accelerators."""
+    if accelerators is None:
+        if instance_type.startswith('a2-highgpu-'):
+            # NOTE: While it is allowed to use A2 machines as CPU-only nodes,
+            # we exclude this case as it is uncommon and undesirable.
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'A2 instance types should be used with A100 GPUs. '
+                    'Either use other instance types or specify the '
+                    'accelerators as A100.')
+        return
+
+    acc = list(accelerators.items())
+    assert len(acc) == 1, acc
+    acc_name, acc_count = acc[0]
+
+    if 'tpu' in acc_name.lower():
+        # TODO(woosuk): Add validation for TPUs.
+        return
+    acc_name = acc_name.upper()
+
+    # Treat A100 as a special case.
+    if acc_name == 'A100':
+        # A100 must be attached to A2 instance type.
+        if not instance_type.startswith('a2-highgpu-'):
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    f'A100 GPUs cannot be attached to {instance_type}. '
+                    'Use A2 instance type instead.')
+        a100_instance_type = _A100_INSTANCE_TYPES[acc_count]
+        if instance_type != a100_instance_type:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    f'A100:{acc_count} cannot be attached to {instance_type}. '
+                    f'Use {a100_instance_type} instead.')
+        return
+
+    # Other GPUs must be attached to N1 machines.
+    # Refer to: https://cloud.google.com/compute/docs/machine-types#gpus
+    if not instance_type.startswith('n1-'):
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(
+                f'{acc_name} GPUs cannot be attached to {instance_type}. '
+                f'Use N1 instance types instead.')
```

Comment:
Shall we add L452 "Refer to: url" here too? Same comment for all other raise's in this func.

Reply:
Added.